### PR TITLE
Adds a tool to find Unquoted List Associations

### PR DESCRIPTION
**Pull request by RemieRichards**
## Explanation:
```list("key" = value)``` is a very common, useful pattern in DM, it's a list association.
However, DM has a "feature" where if the key side of a key/val association is a valid identifier (var name) and doesn't have quotes, it'll accept it anyway, eg:
```list(key = value)``` is still just as valid! however, what if key was a variable?
```
var/key = "woot"
var/list/L = list(key = 5)
```

Are you expecting ```list("woot" = 5)```? I know MSO was, I know I was, I know most of Coderbus was, but thanks to this "feature", it's actually ```list(key = 5)```, the variable name! -fun-

Based on another python text parser I added before, ColonCatcher, except it's regex based, so it really only reuses the (boilerplate) file walking and log outputting.

Also includes an example log, like I did with ColonCatcher.
If you look at this example log, you'll quite quickly be able to work out that almost all (like, all-2) are from the ```armor``` lists, and who touched ```armor``` lists in a major way recently? @phil235 ! and this is why copypasta is bad kids.

## Stats:
<B>146</B>/2012 files have Unquoted List Associations in them
There are <B>2673</B> total Unquoted List Associations (and like 2671 of them were probably by Phil :P)
Found 3 erroneous outputs in the log, out of 2673, that's <B>99.88%</B>(99.887766554433221099887766554433%) accurate

### CODE REVIEW

File: tools/UnquotedListAssociations/2016-11-27-unquoted_list_associations_log.txt [hunk 1]
```diff
@@ -0,0 +1,2967 @@
...
+list(bullet = 70)
+list(energy = 0)
+list(bomb = 60)
+list(bio = 0)
+list(rad = 0)
+list(fire = 100)
+list(acid = 100)
+Total Unquoted List Associations: 21
+..code\game\gamemodes\clock_cult\clock_scriptures\scripture_revenant.dm 
+list(time = 88)
```
Comment (ChangelingRain) [inline tools/UnquotedListAssociations/2016-11-27-unquoted_list_associations_log.txt]: this isn't correctly-found; the line in question is `animate(invoker, color = list(rgb(255, 255, 255), rgb(255, 255, 255), rgb(255, 255, 255), rgb(0,0,0)), time = 88)` \, putting the found line outside of the list.

Comment (RemieRichards) [inline tools/UnquotedListAssociations/2016-11-27-unquoted_list_associations_log.txt]: true, quite complex to fix I believe however.

**Comment by Cyberboss** [comment]
Oh no, RIP my ENTIRE FUCKING ALARM REFACTOR!!!!

**Comment by Supermichael777** [comment]
seems like is a problem with your expectations seeing as key was set to = 5

What do i know though? ¯\_(ツ)_/¯

**Comment by optimumtact** [comment]
Wait does this mean we have like a tonne of bugs?

**Comment by PKPenguin321** [comment]
>Wait does this mean we have like a tonne of bugs?

No, /tg/ is bug free

**Pull request merged**